src/Compilers/Core/Portable/Operations/OperationWalker.cs

```diff
@@ -54,4 +54,57 @@ internal override void VisitNoneOperation(IOperation operation)
             VisitArray(operation.Children);
         }
     }
+
+    /// <summary>
+    /// Represents a <see cref="OperationVisitor{TArgument, TResult}"/> that descends an entire <see cref="IOperation"/> tree
+    /// visiting each IOperation and its child IOperation nodes in depth-first order. Returns null.
+    /// </summary>
+    public abstract class OperationWalker<TArgument> : OperationVisitor<TArgument, object?>
+    {
+        private int _recursionDepth;
+
+        internal void VisitArray<T>(IEnumerable<T> operations, TArgument argument) where T : IOperation
```

Comment:
>VisitArray [](start = 22, length = 10)

Usage of "Array" in the name of this method is somewhat confusing. Consider renaming to "VisitOperations", "VisitSequence" or something similar

Reply:
I'm actually restructuring these in my `Children` refactoring PR (coming a bit later), and have already removed the `VisitOperationArrayElement` methods and renamed this method. I'll leave these until then.

---
In reply to: [526900387](https://github.com/dotnet/roslyn/pull/49453#discussion_r526900387) [](ancestors = 526900387)